### ISSUE TITLE
AutoFixture integration

### ISSUE THREAD

**Issue by sergeyshushlyapin** [sergeyshushlyapin/Sitecore.FakeDb#43]
Should be possible to deserialize items using `AutoData` attribute:

```csharp
public class AutoFixtureTest
{
  [Theory, AutoDb]
  public void ShouldDeserializeItems(Db db)
  {
    // act
    var item = db.GetItem("/sitecore/content/home");

    // assert
    item.Should().NotBeNull();
    item["Title"].Should().Be("Sitecore");
  }
}

public class AutoDbAttribute : AutoDataAttribute
{
  public AutoDbAttribute()
    : base(new Fixture().Customize(new AutoDbCustomization()))
  {
  }
}

public class AutoDbCustomization : ICustomization
{
  public void Customize(IFixture fixture)
  {
    fixture.Customize<Db>(d => d.Do(x => x.Add(new DsDbItem("/sitecore/content/home"))));
  }
}
```

**Comment by sergeyshushlyapin** [comment]
See no way to dispose the `db` which is critical. That's weird...  
https://github.com/AutoFixture/AutoFixture/issues/17

**Comment by sergeyshushlyapin** [comment]
AutoFixture is about *automating non-relevant Test Fixture Setup*. That is not something applicable to the concrete item deserialization.
The Xunit's `DataAttribute` is the right way to go.

**Comment by sergeyshushlyapin** [comment]
I was wrong about assigning the `invalid` label. It *does* may sense to integrate with the @AutoFixture. The integration should not deal with item serializations as I originally thought. It should 'concentrate' on the test data generating:
> Its primary goal is to allow developers to focus on what is being tested rather than how to setup the test scenario

In case of Sitecore, FakeDb should *help* AutoFixture to 'generate' complex types such as `Item`, `Database` or even `PipelineArgs` (yes, you won't be able to generate it easily without FakeDb):

```csharp
[Theory, AutoDbData]
public void ShouldCreateContentItem([Content]Item item)
{
  // act
  var result = Database.GetItem(item.Uri);

  // assert
  result.Should().NotBeNull();
}
```

**Comment by sergeyshushlyapin** [comment]
Released as 0.29.0
